applications/StructuralMechanicsApplication/custom_constitutive/elastic_isotropic_3d.cpp

```diff
@@ -219,19 +219,18 @@ Vector& ElasticIsotropic3D::CalculateValue(
         const bool flag_const_tensor = r_flags.Is( ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR );
         const bool flag_stress = r_flags.Is( ConstitutiveLaw::COMPUTE_STRESS );
 
-        // Set flags to only compute the strain
-        r_flags.Set(ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR, false);
-        r_flags.Set(ConstitutiveLaw::COMPUTE_STRESS, false);
+        ConstitutiveLaw::StrainVectorType& r_strain_vector = rParameterValues.GetStrainVector();
+
+        if( r_flags.IsNot( ConstitutiveLaw::USE_ELEMENT_PROVIDED_STRAIN )) {
+            //Since we are in small strains, any strain measure works, e.g. CAUCHY_GREEN
+            CalculateCauchyGreenStrain(rParameterValues, r_strain_vector);
+        }
+        AddInitialStrainVectorContribution<StrainVectorType>(r_strain_vector);
 
-        ElasticIsotropic3D::CalculateMaterialResponsePK2(rParameterValues);
```

Comment:
This call was only to compute the strain vector, so I just took that part from `CalculateMaterialResponsePK2`. If everyone is ok with this change, I think it would probably be better to encapsulate the new lines in a new function to avoid duplication. (suggestions for names?)